### PR TITLE
posix_spawn: create new process group for child process.

### PR DESCRIPTION
**Pull request by joakim-hove**
**Task**
Make sure spawned out processes get their own pgroup.

**Pre un-WIP checklist**
- [ ] Statoil tests pass locally

### CODE REVIEW

**Comment by pgdr** [comment]
Shouldn't `spawn` be moved to res while we're at it?

**Comment by pgdr** [comment]
And shouldn't we stop using `__varname` in C?  For instance, isn't `__argv` some preserved runtime variable?

**Comment by joakim-hove** [comment]
> Shouldn't spawn be moved to res while we're at it?

OK - will do.

> And shouldn't we stop using __varname in C? 

Good point - I think `__symbol` is reserved in the sense that the standard library might introduce it as a conflicting symbol any day.

**Comment by joakim-hove** [comment]
> Shouldn't spawn be moved to res while we're at it?

I started the process, but the traceback in the `util_abort( )` function is based on calling the `addr2line` program with the `util_spawn( )` function, would therefor at least like to wait with this removal.

**Comment by pgdr** [comment]
Well, `util_abort` doesn't belong in the public API of ecl either, but I agree that we can wait with moving it out.

File: lib/util/util_spawn.c [hunk 1]
```diff
@@ -20,13 +20,13 @@ extern char **environ;
 #define STDERR_FILENO 2
 
 
-static void __init_attributes(posix_spawnattr_t * attributes) {
+static void spawn_init_attributes__(posix_spawnattr_t * attributes) {
```
Comment (pgdr) [inline lib/util/util_spawn.c]: I would guess that if you absolutely need to annotate the function names, then _one_ trailing underscore would suffice?

**Comment by joakim-hove** [comment]
> Well, util_abort doesn't belong in the public API of ecl either,

No - that might be true; but the backtrace capabilities of `util_abort(...)` have in the past proved tremendously valuable when debugging. The bug rate in libecl is not very high - but it still feels a bit brutal to say good bye to the backtrace possibility.

**Comment by pgdr** [comment]
I think this patch should have been reviewed before merging.

**Comment by joakim-hove** [comment]
> I think this patch should have been reviewed before merging.

Well; I thought you had reviewed it?

**Pull request merged**